src/context/navbarMenuContext.tsx

```diff
@@ -0,0 +1,87 @@
+import { createContext, useEffect, useReducer, useState } from "react";
+import { propsModel } from "../model/propsContextModel";
+import { apiGetCategories, categoriesModel } from "../api/apiGetCategories";
+import { REDUCER_ACTION_TYPE, initState, reducer } from "../hooks/navbarHooks";
+import { useMediaQuery } from "@mui/material";
+
+export interface contextModel {
+  stepperTextArr: stepperObjModel[];
+  toggleSearchButton: (value: boolean) => void;
+  handleNext: () => void;
+  handlePrevious: () => void;
+  matches: boolean;
+  handleMenuBurger: () => void;
+  categories: categoriesModel[] | undefined;
```

Comment:
If a prop can be something or undefined then use optional operator maybe? `categories ?: categoriesModel[]`